Dart/thirdPartySrc/analysisServer/com/google/dart/server/internal/remote/RemoteAnalysisServerImpl.java

```diff
@@ -29,7 +29,15 @@
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonPrimitive;
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationType;
+import com.intellij.notification.Notifications;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.jetbrains.lang.dart.DartBundle;
```

Comment:
Is it possible to keep separation of Dart Analysis Server concepts and IntelliJ-specific implementation?

I have 2 ideas.

The easiest would probably be to handle requests from the server in `processNotification()` and pass them to `AnalysisServerListener`. We have `AnalysisServerListener` implementation in `DartAnalysisServerService`, it will need to handle requests from the server.

Alternatively, it might be possible to introduce some new interface like `AnalysisServerRequestsHandler`, and pass requests to it. The client (`DartAnalysisServerService`) will need to register and implement such handler.

Reply:
Abstracting things out and refactoring isn't bad, but I didn't want to over-architect such a small contribution.